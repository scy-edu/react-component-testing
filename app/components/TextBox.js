'use strict'

import React, { Component } from 'react'
import PropTypes from 'prop-types';
import {
  TextInput,
  StyleSheet,
} from 'react-native'

export const style = StyleSheet.create({
  default: {
    backgroundColor: 'grey',
  },
  primary: {
    backgroundColor: 'blue',
  },
});

export default class TextBox extends Component {
  state = {
    value: '',
  };

  _onChange = (value) => {
    this.setState({
      value
    });
  };

  render() {
    const {
      type,
    } = this.props;

    const {
      value,
    } = this.state;

    return (
      <TextInput
        onChange={ this._onChange }
        style={ style[type] }
        value={ value } />
    );
  }

};

TextBox.propTypes = {
  type: PropTypes.oneOf(['default', 'primary']).isRequired,
};

TextBox.defaultProps = {
  type: 'default',
};

// debugging purposes
TextBox.displayName = 'TextBox'
