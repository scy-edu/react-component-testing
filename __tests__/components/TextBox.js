'use strict';

import {
  TextInput,
} from 'react-native';
import React from 'react';
import TextBox, {
  style,
} from '../../app/components/TextBox';

import renderer from 'react-test-renderer';
import {
  shallow,
} from 'enzyme';

// create test props
const createTestProps = () => ({
  type: 'default',
});

describe('<TextBox />', () => {
  describe('rendering', () => {

    let wrapper,
        props = createTestProps();

    beforeEach(() => {
      wrapper = shallow(<TextBox { ...props } />);
    });

    // snapshot
    it('renders correctly', () => {
      const tree = renderer.create(
        <TextBox { ...props } />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render TextInput', () => {
      expect(wrapper.find('TextInput')).toHaveLength(1);
      expect(wrapper.type()).toBe(TextInput);
    });

    it('should have default styling', () => {
      expect(wrapper.find('TextInput').prop('style')).toBe(style.default);
    });

    describe('primary styles', () => {
      beforeEach(() => {
        wrapper = shallow(
          <TextBox type='primary' />
        );
      });

      it('should have primary style', () => {
        expect(wrapper.find('TextInput').prop('style')).toBe(style.primary);
      });

    });

  });

  describe('interactions', () => {
    let wrapper,
        props = createTestProps(),
        newValue = 'Hello world';

    beforeEach(() => {
      wrapper = shallow(
        <TextBox { ...props } />
      );
    });

    it('should have a blank initial state', () => {
      expect(wrapper.state('value')).toBe('');
    });

    it('should be able to change the value based on typing', () => {
      wrapper.setState({
        value: newValue,
      });

      expect(wrapper.state('value')).toBe(newValue);
    });

    it('should change the state when the input changes', () => {
      wrapper.simulate('change', newValue);
      expect(wrapper.state('value')).toBe(newValue);
    });
  });

});
