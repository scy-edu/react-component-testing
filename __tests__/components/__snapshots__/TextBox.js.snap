// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBox /> rendering renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChange={[Function]}
  style={
    Object {
      "backgroundColor": "grey",
    }
  }
  value=""
/>
`;
